x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/details_page/components/settings/index.tsx

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { memo, useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+import { EuiBottomBar, EuiFlexGroup, EuiFlexItem, EuiButtonEmpty, EuiButton } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { AGENT_CONFIG_PATH } from '../../../../../constants';
+import { AgentConfig } from '../../../../../types';
+import {
+  useCore,
+  useCapabilities,
+  sendUpdateAgentConfig,
+  useConfig,
+  sendGetAgentStatus,
+} from '../../../../../hooks';
+import {
+  AgentConfigForm,
+  agentConfigFormValidation,
+  ConfirmDeployConfigModal,
+} from '../../../components';
+import { useConfigRefresh } from '../../hooks';
+
+const FormWrapper = styled.div`
+  max-width: 800px;
+  margin-right: auto;
+  margin-left: auto;
+`;
+
+export const ConfigSettingsView = memo<{ config: AgentConfig }>(
+  ({ config: originalAgentConfig }) => {
+    const {
+      notifications,
+      chrome: { getIsNavDrawerLocked$ },
+    } = useCore();
+    const {
+      fleet: { enabled: isFleetEnabled },
+    } = useConfig();
+    const history = useHistory();
+    const hasWriteCapabilites = useCapabilities().write;
+    const refreshConfig = useConfigRefresh();
+    const [isNavDrawerLocked, setIsNavDrawerLocked] = useState(false);
+    const [agentConfig, setAgentConfig] = useState<AgentConfig>({
+      ...originalAgentConfig,
```

Comment:
Does this need to be a copy of the object?

Reply:
not really, just a personal habit when working with objects that are passed by reference (though in this context with useState it doesn't matter much)